docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+* The service cannot provide adequate information to produce a full specification
+* Unforeseen limitations within ACK or the service directly
+
+For any of these cases, the resource can be marked as “not adoptable” and the validation webhook will subsequently reject any adopted resources of this type. Therefore, the manager will never attempt to run the adoption process. This flag can be set in the generator configuration file with the `cannot_adopt` boolean option supplied for any resource.
+
+## Adoption Process Diagram
+
+![sequence-diagram](./images/adoption-sequence-diagram.png)
+# Potential Alternatives
+
+## Resource Adoption Annotation
+
+The existing custom resource definitions (every service resource) would accept an additional annotation that the manager interprets as an indication that the resource specification should be ignored and instead that it should be formed through adoption. For any resource marked with this annotation, upon creation, the manager would attempt to describe the AWS resource associated with the annotated value. If none such resource exists, the manager would update the state of the custom resource into a `AdoptionFailed` terminal state. If the resource does exist, the manager would continue the follow the same adoption process as defined in the [Proposed Implementation](#proposed-implementation) section.
+
+### Example Specification
+
+**An example adopted S3 resource**
+
+```
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: Bucket
+metadata:
+  annotations:
+    services.k8s.aws/arn: arn:aws:s3:::my-existing-bucket
+  ...
+spec:
+   ...
+```
+
+### Notable Pitfalls
+
+In the event that the user is attempting to restore resources created in a previous installation of ACK (in the event of an etcd corruption, for example), each manifest will need to be updated with the adoption annotation. Therefore the restoration process would require modifying desired state files (k8s manifests) with actual state fields (the created ARN/ID fields). At the least this introduces a temporary change in k8s manifests - add ARNs to manifests,  apply and adopt, delete ARNs from manifests. More so, it could introduce private field information (generated public API IDs, RDS instance ARNs, etc.) into the manifests repository.
+
+Another pitfall is that, given this solution aims to re-use the existing custom resource definitions, all adopted resources would need to provide a specification body that contains all required fields. Every service contains a distinct set of required values, and as such the user would need to modify every adopted resource type with a distinct set of valid dummy values. Assuming a user provides dummy values for these fields, they would subsequently be overwritten by the manager logic when updating the resource.
+
+# In Scope
+
+* Declaration of the new `AdoptedResource` CRD
+* Updates to generator for service-specific configuration
+* Code paths for service managers to receive informer updates
```

Comment:
hmmm, actually you may not need this. Since all resource managers implement the ReadOne and Create methods, the common ACK runtime can handle all the Informer logic (see `pkg/runtime/cache` for how it handles Informers for Namespaces and ConfigMaps currently). Let's discuss implementation details a little later. I have thoughts here... :)

Reply:
Yeah I had that same idea, I suppose with this line I meant indirectly handle `AdoptedResource` updates.